### PR TITLE
test: some more skip tests

### LINKED ISSUE #1532: More TX_SKIP edge cases
# `TX_SKIP` message calls

The spec says that one should keep warmths for `TX_SKIP` as they are. I wonder what that would mean e.g. if
- sender is a precompile (does our testing environment even allow this ? or would we need to produce a _signature_ for the (unknown) private key of a precompile) ?
- coinbase is a precompile (those are warm by default)

**N.B.** Recipients may not be precompiles.

source https://github.com/Consensys/linea-tracer/pull/1510#issuecomment-2491185905

# `TX_SKIP` deployment transactions

If we don't have them yet we should add tests for TX_SKIP with trivial deployments. We can play the same games of
- sender is coinbase
- (deployment address) is coinbase
- coinbase is a precompile

source https://github.com/Consensys/linea-tracer/pull/1510#discussion_r1852065538

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**

### CODE REVIEW

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java [hunk 1]
```diff
@@ -112,14 +112,14 @@ public void resolveAtEndTransaction(
     // may have to be modified in case of address collision
     senderNew = canonical(hub, world, sender.address(), isPrecompile(sender.address()));
     recipientNew = canonical(hub, world, recipient.address(), isPrecompile(recipient.address()));
-    coinbaseNew = canonical(hub, world, coinbase.address(), isPrecompile(recipient.address()));
+    coinbaseNew = canonical(hub, world, coinbase.address(), isPrecompile(coinbase.address()));
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java:115]: bug

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java [hunk 2]
```diff
@@ -135,6 +135,9 @@ public void resolveAtEndTransaction(
       if (recipientIsCoinbase()) {
         recipientNew = coinbaseNew.deepCopy().decrementBalanceBy(txMetadata.getCoinbaseReward());
         recipient = recipientNew.deepCopy().decrementBalanceBy(value);
+        if (txMetadata.isDeployment()){
```
Comment (letypequividelespoubelles) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java]: bug fix

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java (hunk 1 above)
Comment (OlivierBBB) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java:115]: Indeed

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java [hunk 3]
```diff
@@ -135,6 +135,9 @@ public void resolveAtEndTransaction(
       if (recipientIsCoinbase()) {
         recipientNew = coinbaseNew.deepCopy().decrementBalanceBy(txMetadata.getCoinbaseReward());
         recipient = recipientNew.deepCopy().decrementBalanceBy(value);
+        if (txMetadata.isDeployment()) {
+          recipient.decrementNonceByOne().decrementDeploymentNumberByOne();
+        }
```
Comment (OlivierBBB) [inline arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java:140]: Interesting catch. I take it this blew up in one of the `deploymentCoinbaseIsEffectiveTo`-type tests ?

File: arithmetization/src/test/java/net/consensys/linea/zktracer/module/hub/TxSkipTests.java [hunk 4]
```diff
@@ -320,4 +321,122 @@ void senderIsCoinbaseIsReceiver() {
         .build()
         .run();
   }
+
+  @Test
```
Comment (OlivierBBB) [inline arithmetization/src/test/java/net/consensys/linea/zktracer/module/hub/TxSkipTests.java:325]: These should be made parametric and again test everything that is possible.

Comment (OlivierBBB) [reply arithmetization/src/test/java/net/consensys/linea/zktracer/module/hub/TxSkipTests.java:325]: Also making the tests parametric results in fewer lines of code.

**Review by OlivierBBB — CHANGES_REQUESTED** [review]
Parametric tests

File: arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java (hunk 3 above)
Comment (letypequividelespoubelles) [reply arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java:140]: yes

**Review by OlivierBBB — APPROVED** [review]
To be made parametric.
